src/priceHistory/slice.ts

```diff
@@ -0,0 +1,68 @@
+import { Draft, PayloadAction, createSlice } from '@reduxjs/toolkit'
+import BigNumber from 'bignumber.js'
+
+type PriceHistoryStatus = 'idle' | 'loading' | 'success' | 'error'
+
+export interface Price {
+  priceUsd: BigNumber.Value
+  priceFetchedAt: number
+}
+
+export interface State {
+  [tokenId: string]: {
+    status: PriceHistoryStatus
+    prices: Price[]
+  }
+}
+
+const slice = createSlice({
+  name: 'priceHistory',
+  initialState: {},
+  reducers: {
+    fetchPriceHistoryStart: (
+      state: Draft<State>,
+      action: PayloadAction<{
+        tokenId: string
+        startTimestamp: number
+        endTimestamp: number
+      }>
+    ) => {
+      const { tokenId } = action.payload
+      state[tokenId] = {
+        ...state[tokenId],
+        status: 'loading',
+      }
+    },
+    fetchPriceHistorySuccess: (
+      state: Draft<State>,
+      action: PayloadAction<{
+        tokenId: string
+        prices: Price[]
+      }>
+    ) => {
+      const { tokenId, prices } = action.payload
+      const token = state[tokenId]
+      if (token) {
```

Comment:
and what if the token isn't there? seems like it would be good to still include the updated state. I think I left a similar comment on your last PR, though I've lost the link for that so idk what the outcome of the discussion was

Reply:
The token will be there because of the previous fetchPriceHistoryStart, but maybe it's better the handle similar to that?

```
state[tokenId] = {
  ...state[tokenId],
  status: 'success',
  prices,
}
```